saw-core-aig/src/SAWCoreAIG/BitBlast.hs

```diff
@@ -475,17 +475,14 @@ bitBlastExtCns ecMap (EC (VarName idx name) _v) =
       "SAWCoreAIG.BitBlast: can't translate variable " ++
       show name ++ "(index: " ++ show idx ++ ")"
 
-asAIGType :: SharedContext -> Term -> IO [(String, Term)]
-asAIGType sc t = do
-  t' <- scWhnf sc t
-  case t' of
-    (R.asPi -> Just (n, t1, t2)) -> ((Text.unpack n, t1) :) <$> asAIGType sc t2
-    (R.asBoolType -> Just ())    -> return []
-    (R.asVecType -> Just _)      -> return []
-    (R.asTupleType -> Just _)    -> return []
-    (R.asRecordType -> Just _)   -> return []
-    _                            -> fail $ "SAWCoreAIG.BitBlast.adAIGType: invalid AIG type: "
-                                    ++ scPrettyTerm PPS.defaultOpts t'
+asPiTypes :: SharedContext -> Term -> IO ([(String, Term)], Term)
+asPiTypes sc t =
+  do t' <- scWhnf sc t
+     case R.asPi t' of
+       Nothing -> pure ([], t)
+       Just (n, t1, t2) ->
+         do (args, ret) <- asPiTypes sc t2
+            pure ((Text.unpack n, t1) : args, ret)
```

Comment:
I wonder if it would be worth factoring this bit out into a `asPiList :: Term -> ([(LocalName, Term)], Term)` function and placing it alongside [`asLambdaList`](https://github.com/GaloisInc/saw-script/blob/3d1dfceea586e513b670809563981a68dbfcd129/saw-core/src/SAWCore/Recognizer.hs#L316-L319) in `saw-core:SAWCore.Recognizer`.

Reply:
The Recognizer library already has `asPiList`:
https://github.com/GaloisInc/saw-script/blob/3d1dfceea586e513b670809563981a68dbfcd129/saw-core/src/SAWCore/Recognizer.hs#L325-L330

The function here is a bit different, though, in that it calls `scWhnf` to normalize the term at each step.